recipes/key-cache-advanced.md

```diff
@@ -0,0 +1,108 @@
+
+Examples of using `checksum`:
+- `cache-key-{{ checksum "package-lock.json" }}`
+- `cache-key-{{ checksum "**/Package.resolved" }}`
+- `cache-key-{{ checksum "**/*.gradle*" "gradle.properties" }}`
+
+`getenv`: This function returns the value of an environment variable or an empty string if the variable is not defined.
+
+Examples of `getenv`:
+- `cache-key-{{ getenv "PR" }}`
+- `cache-key-{{ getenv "BITRISEIO_PIPELINE_ID" }}`
+
+## Skip saving the cache in PR builds (restore only)
+
+If you want builds triggered by pull requests to only restore the cache and skip saving it, you can run the **Save cache** Step conditionally:
+
+```yaml
+steps:
+- restore-cache@1:
+    inputs:
+    - key: node-modules-{{ checksum "package-lock.json" }}
+
+# Build steps
+
+- save-cache@1:
+    run_if: ".IsCI | and (not .IsPR)" # Condition that is false in PR builds
+    inputs:
+    - key: node-modules-{{ checksum "package-lock.json" }}
+    - paths: node_modules
+```
+
+## Separate caches for each OS and architecture
+
+Cache is not guaranteed to work across different Bitrise Stacks (different OS or same OS but different CPU architecture). If a Workflow runs on different stacks, it's a good idea to include the OS and architecture in the **Cache key** input:
+
+```yaml
+steps:
+- save-cache@1:
+    inputs:
+    - key: '{{ .OS }}-{{ .Arch }}-npm-cache-{{ checksum "package-lock.json" }}'
+    - path: node_modules
+```
+
+## Multiple independent caches
+
+You can add multiple instances of the cache steps to a Workflow:
+
+```yaml
+steps:
+- save-cache@1:
+    title: Save NPM cache
+    inputs:
+    - paths: node_modules
+    - key: npm-cache-{{ checksum "package-lock.json" }}
+- save-cache@1:
+    title: Save Python cache
+    inputs:
+    - paths: venv/
+    - key: pip-cache-{{ checksum "requirements.txt" }}
+```
+
+## Cache warm-up for pull requests
+
+Caching works the best when the cached content is up to date and contains useful data for dependency managers and build systems. It's a good idea to run a workflow periodically that builds the project from the latest code on the main branch and saves the result in the cache. This way other builds triggered by pull requests can restore an up-to-date cache.
+
+By including a checksum in the cache key, the **Save cache** step will save multiple unique cache archives when the project files change (instead of overriding the previous cache). This way PRs not targeting the latest state of the main branch can still download a relevant cache archive.
+
+```yaml
+workflows:
+    pr-validation:
+        steps:
+        - restore-cache@1:
+            inputs:
+            - key: |-
+                node-modules-{{ checksum "package-lock.json" }}
+                node-modules-
+        # Rest of the PR validation workflow
+    
+    cache-warm-up:
+        description: This workflow should be triggered periodically or by a push event on the main branch
```

Comment:
```suggestion
        description: This Workflow should be triggered periodically or by a push event on the main branch
```

Reply:
I'm not entirely sure what this sentence means. (Do you mean that this should either be run manually on a regular basis, or conditionally attached to a trigger map that fires when a code push happens?)  
I would propose something like: 
This Workflow should either be run on a scheduled basis or triggered by a push event on the main branch periodically.